[Bug] r.pi modules do not compile

**Name of the addon**
r.pi

**Describe the bug**
Compilation fails Windows, see e.g. [log here](https://wingrass.fsv.cvut.cz/grass79/x86_64/addons/grass-7.9.dev/logs/r.pi.log)

Additionally, `g.extension r.pi` fails on Ubuntu with the following although r.pi can be compiled directly.
```
Fetching <r.pi> from GRASS GIS Addons repository (be patient)...
Compiling...
matrix.c: In function ‘writeDistMatrixAndID’:
matrix.c:42:20: warning: format not a string literal and no format arguments [-Wformat-security]
      G_fatal_error(msg);
                    ^~~
matrix.c: In function ‘writeDistMatrixAndID’:
matrix.c:44:20: warning: format not a string literal and no format arguments [-Wformat-security]
      G_fatal_error(msg);
                    ^~~
OBJ.x86_64-pc-linux-gnu/heap.o: In function `upheap':
/tmp/grass7-anna-15561/tmp2zi1v7ur/r.pi/r.pi.fnn/heap.c:19: undefined reference to `exchange'
OBJ.x86_64-pc-linux-gnu/heap.o: In function `downheap':
/tmp/grass7-anna-15561/tmp2zi1v7ur/r.pi/r.pi.fnn/heap.c:36: undefined reference to `exchange'
collect2: error: ld returned 1 exit status
make[1]: *** [/tmp/grass7-anna-15561/tmp2zi1v7ur/r.pi/bin/r.pi.fnn] Error 1
frag.c: In function ‘writeFrag_local’:
frag.c:71:8: warning: ‘x’ is used uninitialized in this function [-Wuninitialized]
     if (x > 0 && flagbuf[y * ncols + x - 1] != 0)
        ^
frag.c:71:28: warning: ‘y’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     if (x > 0 && flagbuf[y * ncols + x - 1] != 0)
                          ~~^~~~~~~
Installing...
make[1]: *** No rule to make target 'install'.  Stop.
/usr/bin/install: cannot stat '/tmp/grass7-anna-15561/tmp2zi1v7ur/r.pi/bin/r.pi.fnn': No such file or directory
make[1]: *** [install] Error 1
Updating extensions metadata file...
Updating extension modules metadata file...
WARNING: No metadata available for module 'r.pi'.
WARNING: No metadata available for module 'r.pi.fnn'.
ERROR: Unable to read manual page: [Errno 2] No such file or directory:
       '/home/anna/.grass7/addons/docs/html/r.pi.fnn.html'
```

It also compiles on Mac OSX 10.15.7 in GRASS7.9dev (binary from Michael from Jan 11), it just gives the same warnings. Helena

GRASS nc_spm_08_grass7/assignments:~ > g.extension r.pi
Fetching <r.pi> from GRASS GIS Addons repository (be patient)...
Compiling...
matrix.c:42:20: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
            G_fatal_error(msg);
                          ^~~
matrix.c:42:20: note: treat the string as an argument to avoid this
            G_fatal_error(msg);
                          ^
                          "%s", 
1 warning generated.
matrix.c:44:20: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
            G_fatal_error(msg);
                          ^~~
matrix.c:44:20: note: treat the string as an argument to avoid this
            G_fatal_error(msg);
                          ^
                          "%s", 
1 warning generated.
Installing...
make[1]: *** No rule to make target `install'.  Stop.
Updating extensions metadata file...
Updating extension modules metadata file...
WARNING: No metadata available for module 'r.pi'.
Installation of <r.pi> successfully finished`

--- a/grass7/raster/r.pi/r.pi.enn/matrix.c
+++ b/grass7/raster/r.pi/r.pi.enn/matrix.c
@@ -35,11 +35,8 @@ int writeDistMatrixAndID(char *name, Coords ** frags, int count)
 	/* open the new cellfile */
 	out_fd = Rast_open_new(name, DCELL_TYPE);
 	if (out_fd < 0) {
-	    char msg[200];
-
-	    sprintf(msg, "can't create new cell file <%s> in mapset %s\n",
-		    name, mapset);
-	    G_fatal_error(msg);
+	    G_fatal_error(_("can't create new cell file <%s> in mapset %s\n"),
+                      name, G_mapset());
 	    res = 1;
 	}
 	else {

--- a/grass7/raster/r.pi/r.pi.enn/matrix.c
+++ b/grass7/raster/r.pi/r.pi.enn/matrix.c
@@ -4,7 +4,6 @@ int writeDistMatrixAndID(char *name, Coords ** frags, int count)
 {
     FILE *out_fp;
     int res = 0;
-    const char *mapset;
     int out_fd;
     int row, col, i;
     DCELL *result;
@@ -30,7 +29,6 @@ int writeDistMatrixAndID(char *name, Coords ** frags, int count)
 	/* check if the new file name is correct */
     if (G_legal_filename(name) < 0)
 	    G_fatal_error(_("<%s> is an illegal file name"), name);
-	mapset = G_mapset();
 
 	/* open the new cellfile */
 	out_fd = Rast_open_new(name, DCELL_TYPE);

--- a/grass7/raster/r.pi/r.pi.fnn/heap.c
+++ b/grass7/raster/r.pi/r.pi.fnn/heap.c
@@ -3,7 +3,7 @@
 #include <grass/stats.h>
 #include "local_proto.h"
 
-inline void exchange(int p1, int p2)
+static void exchange(int p1, int p2)
 {
     Path_Coords tmp = heap[p1];
 

--- a/grass7/raster/r.pi/r.pi.fnn/matrix.c
+++ b/grass7/raster/r.pi/r.pi.fnn/matrix.c
@@ -37,11 +37,8 @@ int writeDistMatrixAndID(char *name, Coords ** frags, int count)
 	/* open the new cellfile */
 	out_fd = Rast_open_new(name, DCELL_TYPE);
 	if (out_fd < 0) {
-	    char msg[200];
-
-	    sprintf(msg, "can't create new cell file <%s> in mapset %s\n",
-		    name, mapset);
-	    G_fatal_error(msg);
+	    G_fatal_error(_("can't create new cell file <%s> in mapset %s\n"),
+                      name, G_mapset());
 	    res = 1;
 	}
 	else {

--- a/grass7/raster/r.pi/r.pi.fnn/matrix.c
+++ b/grass7/raster/r.pi/r.pi.fnn/matrix.c
@@ -4,7 +4,6 @@ int writeDistMatrixAndID(char *name, Coords ** frags, int count)
 {
     FILE *out_fp;
     int res = 0;
-    const char *mapset;
     int out_fd;
     int row, col, i;
     DCELL *result;
@@ -32,7 +31,6 @@ int writeDistMatrixAndID(char *name, Coords ** frags, int count)
 	    G_warning(_("<%s> is an illegal file name"), name);
 	    res = 1;
 	}
-	mapset = G_mapset();
 
 	/* open the new cellfile */
 	out_fd = Rast_open_new(name, DCELL_TYPE);

